### PR TITLE
Docs: review and update connector form scratch documentation and related guides

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

This pull requests is my first walk through for of building a connector from scratch, and cleanup of the related docs:
- Making the docs more clear and concise where possible
- Fixed a few internal links
- Replaced script-based acceptance tests with `airbyte-ci`
- Replaced "Standard tests" with Connector Acceptance Tests

## Recommended reading order
It's probably easiest to review the guides with a quick read. Commit by commit would work, but probably easier in one pass.

## 🚨 User Impact 🚨
Documentation changes only, and focused on the very bare-bones guide. Will not affect Airbyte customers or users.

### CODE REVIEW

File: airbyte-cdk/python/README.md [hunk 1]
```diff
@@ -2,9 +2,9 @@
```
Comment (natikgadzhi) [inline airbyte-cdk/python/README.md:2]: That file's contents are also duplicated as a docs page. I wonder if there's a nice clean way to import a markdown file with a docusaurus-blessed imnport markdown syntax, from outside of it's content directory?

File: airbyte-cdk/python/README.md [hunk 2]
```diff
@@ -31,25 +31,19 @@ See the [concepts docs](docs/concepts/) for a tour through what the API offers.
 
 **HTTP Connectors**:
 
-* [Exchangerates API](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-exchange-rates/source_exchange_rates/source.py)
```
Comment (natikgadzhi) [inline airbyte-cdk/python/README.md:34]: This connector have been rewritten in yaml ☠️

File: airbyte-cdk/python/README.md [hunk 3]
```diff
@@ -61,17 +55,20 @@ pip install -e ".[dev]" # [dev] installs development-only dependencies
...
-* Iterate on the code locally
-* Run tests via `python -m pytest -s unit_tests`
-* Perform static type checks using `mypy airbyte_cdk`. `MyPy` configuration is in `mypy.ini`.
- * Run `mypy <files to check>` to only check specific files. This is useful as the CDK still contains code that is not compliant.
-* The `type_check_and_test.sh` script bundles both type checking and testing in one convenient command. Feel free to use it!
+- Iterate on the code locally
+- Run tests via `python -m pytest -s unit_tests`
+- Perform static type checks using `mypy airbyte_cdk`. `MyPy` configuration is in `mypy.ini`.
+- Run `mypy <files to check>` to only check specific files. This is useful as the CDK still contains code that is not compliant.
+- The `type_check_and_test.sh` script bundles both type checking and testing in one convenient command. Feel free to use it!
```
Comment (natikgadzhi) [inline airbyte-cdk/python/README.md:62]: Unrelated: I wonder if we should make a pre-commit hook for this?

File: airbyte-cdk/python/README.md [hunk 4]
```diff
@@ -152,8 +159,8 @@ HTTP requests to `localhost:8113/data` should now return the body defined in the
 
 ## Coming Soon
 
-* Full OAuth 2.0 support \(including refresh token issuing flow via UI or CLI\)
-* Airbyte Java HTTP CDK
-* CDK for Async HTTP endpoints \(request-poll-wait style endpoints\)
-* CDK for other protocols
-* Don't see a feature you need? [Create an issue and let us know how we can help!](https://github.com/airbytehq/airbyte/issues/new?assignees=&labels=type%2Fenhancement&template=feature-request.md&title=)
+- Full OAuth 2.0 support \(including refresh token issuing flow via UI or CLI\)
```
Comment (natikgadzhi) [inline airbyte-cdk/python/README.md:162]: This is sus — I think maybe some of these are already implemented and not updated in docs? /cc @evantahler

File: airbyte-integrations/connector-templates/generator/plopfile.js [hunk 5]
```diff
@@ -24,7 +24,6 @@ ${additionalMessage || ""}
 
 module.exports = function (plop) {
   const docRoot = '../../../docs/integrations';
-  const definitionRoot = '../../../airbyte-config-oss/init-oss/src/main/resources';
```
Comment (natikgadzhi) [inline airbyte-integrations/connector-templates/generator/plopfile.js:27]: That is the only code change, but my tooling and search told me this thing is not used.

File: docs/connector-development/cdk-python/README.md [hunk 6]
```diff
@@ -1,44 +1,55 @@
...
 
 :::info
-Over the next few months, the project will only accept connector contributions that are made using the [Low-Code CDK](https://docs.airbyte.com/connector-development/config-based/low-code-cdk-overview) or the [Connector Builder](https://docs.airbyte.com/connector-development/connector-builder-ui/overview).
+Over the next few months, the project will only accept connector contributions that are made using the
+[Low-Code CDK](https://docs.airbyte.com/connector-development/config-based/low-code-cdk-overview) or the
+[Connector Builder](https://docs.airbyte.com/connector-development/connector-builder-ui/overview).
 
-Contributions made with the Python CDK will be closed, but we will inquire to understand why it wasn't done with Low-Code/Connector Builder so we can address missing features.
-This decision is aimed at improving maintenance and providing a larger catalog with high-quality connectors.
+New pull requests made with the Python CDK will be closed, but we will inquire to understand why it wasn't done with
```
Comment (natikgadzhi) [inline docs/connector-development/cdk-python/README.md:8]: /cc @marcosmarxm, I've tried to clarify the wording here a little bit, but that is your domain! Let me know if I should roll any of this back.

File: docs/connector-development/tutorials/build-a-connector-the-hard-way.md [hunk 7]
```diff
@@ -52,53 +44,51 @@ Here's the outline of what we'll do to build our connector:
...
+work together. For real connectors, the generator provides `Python` and `Python HTTP API` source templates, they use
+[Airbyte CDK](../cdk-python/README.md).
+:::
 
-Head to the connector directory and we should see the following files have been generated:
 
 ```bash
 $ cd ../../connectors/source-stock-ticker-api
 $ ls
-Dockerfile                 README.md                  acceptance-test-config.yml acceptance-test-docker.sh  build.gradle
```
Comment (natikgadzhi) [inline docs/connector-development/tutorials/build-a-connector-the-hard-way.md:97]: Tested locally

File: docs/connector-development/tutorials/build-a-connector-the-hard-way.md [hunk 8]
```diff
@@ -52,53 +44,51 @@ Here's the outline of what we'll do to build our connector:
...
-
 Airbyte provides a code generator which bootstraps the scaffolding for our connector. Let's use it by running:
 
 ```bash
 $ cd airbyte-integrations/connector-templates/generator
 $ ./generate.sh
 ```
 
-We'll select the `generic` template and call the connector `stock-ticker-api`:
+Select the `Generic Source` template and call the connector `stock-ticker-api`:
```
Comment (natikgadzhi) [inline docs/connector-development/tutorials/build-a-connector-the-hard-way.md:75]: Updated to match actual generator options

File: docs/connector-development/tutorials/build-a-connector-the-hard-way.md [hunk 9]
```diff
@@ -52,53 +44,51 @@ Here's the outline of what we'll do to build our connector:
...
 
-We'll start the process from the Airbyte repository root:
+Start the process from the Airbyte repository root:
 
 ```bash
 $ pwd
 /Users/sherifnada/code/airbyte
 ```
 
-First, let's create a new branch:
```
Comment (natikgadzhi) [inline docs/connector-development/tutorials/build-a-connector-the-hard-way.md:72]: Not technically required — I've tried to make the guide as concise as possible.

File: docs/connector-development/tutorials/build-a-connector-the-hard-way.md [hunk 10]
```diff
@@ -153,16 +144,14 @@ Let's create a [JSONSchema](http://json-schema.org/) file `spec.json` encoding t
 - `description` will be shown in the Airbyte UI under each field to help the user understand it
 - `airbyte_secret` used by Airbyte to determine if the field should be displayed as a password \(e.g: `********`\) in the UI and not readable from the API
 
-We'll save this file in the root directory of our connector. Now we have the following files:
 
 ```bash
```
Comment (natikgadzhi) [inline docs/connector-development/tutorials/build-a-connector-the-hard-way.md:148]: Ditto, tested locally

File: docs/connector-development/tutorials/build-a-connector-the-hard-way.md [hunk 11]
```diff
@@ -1017,36 +969,26 @@ acceptance_tests:
 #      configured_catalog_path: "integration_tests/configured_catalog.json"
 #      future_state_path: "integration_tests/abnormal_state.json"
 ```
+To run the test suite, we'll use [`airbyte-ci`](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md).
```
Comment (natikgadzhi) [inline docs/connector-development/tutorials/build-a-connector-the-hard-way.md:972]: /cc @evantahler your PR took 5 minutes from my question on Slack, mine — closer to two weeks, but I got to cleaning it up. ;)

File: airbyte-cdk/python/README.md (hunk 4 above)
Comment (evantahler) [reply airbyte-cdk/python/README.md:162]: 👍 on removing this.

Async is in-the works, and talking about Java doesn't make sense here. We've made some connectors for "other protocols" (SOAP), but I don't think folks really care about the internals.

File: airbyte-cdk/python/README.md [hunk 12]
```diff
@@ -2,9 +2,9 @@
 
 The Airbyte Python CDK is a framework for rapidly developing production-grade Airbyte connectors. The CDK currently offers helpers specific for creating Airbyte source connectors for:
 
-* HTTP APIs \(REST APIs, GraphQL, etc..\)
-* Singer Taps
-* Generic Python sources \(anything not covered by the above\)
+- HTTP APIs \(REST APIs, GraphQL, etc..\)
+- Singer Taps
+- Generic Python sources \(anything not covered by the above\)
```
[the remark this answers is not in the document]
Comment (marcosmarxm) [reply airbyte-cdk/python/README.md:7]: Honestly I don't think the current CDK supports Singer anymore.

File: airbyte-cdk/python/README.md [hunk 13]
```diff
@@ -152,8 +159,8 @@ HTTP requests to `localhost:8113/data` should now return the body defined in the
...
-* Full OAuth 2.0 support \(including refresh token issuing flow via UI or CLI\)
-* Airbyte Java HTTP CDK
-* CDK for Async HTTP endpoints \(request-poll-wait style endpoints\)
-* CDK for other protocols
-* Don't see a feature you need? [Create an issue and let us know how we can help!](https://github.com/airbytehq/airbyte/issues/new?assignees=&labels=type%2Fenhancement&template=feature-request.md&title=)
+- Full OAuth 2.0 support \(including refresh token issuing flow via UI or CLI\)
+- Airbyte Java HTTP CDK
+- CDK for Async HTTP endpoints \(request-poll-wait style endpoints\)
+- CDK for other protocols
+- Don't see a feature you need? [Create an issue and let us know how we can help!](https://github.com/airbytehq/airbyte/issues/new?assignees=&labels=type%2Fenhancement&template=feature-request.md&title=)
```
Comment (marcosmarxm) [inline airbyte-cdk/python/README.md:166]:
```suggestion
- Don't see a feature you need? [Create an issue and let us know how we can help!](https://github.com/airbytehq/airbyte/discussions/categories/ideas-and-features)
```

We're trying to push people to create Discussion against Issues for features (clear upvote system)

File: docs/connector-development/tutorials/building-a-python-source.md [hunk 14]
```diff
@@ -232,9 +232,9 @@ For a brief overview on the catalog check out [Beginner's Guide to the Airbyte C
...
 
-### Step 8: Set up Standard Tests
+### Step 8: Set up Connector Acceptance Tests (CATs)
 
-The Standard Tests are a set of tests that run against all sources. These tests are run in the Airbyte CI to prevent regressions. They also can help you sanity check that your source works as expected. The following [article](../testing-connectors/connector-acceptance-tests-reference.md) explains Standard Tests and how to run them.
+The Connector Acceptance Tests are a set of tests that run against all sources. These tests are run in the Airbyte CI to prevent regressions. They also can help you sanity check that your source works as expected. The following [article](../testing-connectors/connector-acceptance-tests-reference.md) explains Connector Acceptance Tests and how to run them.
 
 You can run the tests using `./gradlew :airbyte-integrations:connectors:source-<source-name>:integrationTest`. Make sure to run this command from the Airbyte repository root.
```
Comment (marcosmarxm) [inline docs/connector-development/tutorials/building-a-python-source.md:239]: This doesn't work anymore it need to update to use the `airbyte-ci` cli instead, do you mind changing it?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Nice changes!

File: airbyte-cdk/python/README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (natikgadzhi) [reply airbyte-cdk/python/README.md:7]: I had to google singer taps. Okay, I'll clean this up!

File: airbyte-cdk/python/README.md (hunk 13 above)
Comment (natikgadzhi) [reply airbyte-cdk/python/README.md:166]: I think I even seen that in _one_ spot it already leads to discussions! Good catch, I'll clean this up.

File: docs/connector-development/tutorials/building-a-python-source.md (hunk 14 above)
Comment (natikgadzhi) [reply docs/connector-development/tutorials/building-a-python-source.md:239]: Yes! I'll get to this — there are multiple spots where this script is still mentioned, and I'll clean it up too.

Comment (natikgadzhi) [reply docs/connector-development/tutorials/building-a-python-source.md:239]: Oh wait, I've already cleaned that one up 🙃

**Comment by natikgadzhi** [comment]
I panic-committed prettier commit (the last one) — I can move it into a separate PR if we want to review, I'm just experimenting and making it easier on my sanity to re-read docs.

**Review by marcosmarxm — APPROVED** [review]

**Comment by natikgadzhi** [comment]
@marcosmarxm thank you <3

**Pull request merged**